src/_blog/rust-libp2p-browser-to-server.md

```diff
@@ -0,0 +1,61 @@
+
+### Benefits of WebRTC over websockets and webtransport
+
+One of the main benefits of using WebRTC on the server is that the it does not need TLS certificates from a DNS domain -- you can run a home server over WebRTC without a dotcom! Instead, with `rust-libp2p` we can use self-signed certificates to provide the secure connections. Websockets however would require TLS certificates.
+
+The second major benefot of WebRTC is it enables (eventual\*) direct browser-to-browser communication. Though webtransport can connect with self-signed certificates, it cannot connect browser-to-browser like WebRTC can.
+
+> \*Note: This is a common point of confusion for WebRTC when connecting two browsers together. Although the browser can _eventually_ connect directly, to set up this direct connection we need a signalling server. In the case of libp2p, we can use a libp2p server to conduct this signalling by using circuit relay. But we always need the WebRTC server first. This server can be any libp2p node that is not running in the browser, it could be a home computer for example.
+
+With WebRTC rust-libp2p now available in the browser, we can enable developers to build full stack applications in rust without the need for a domain name or external server.
+
+### Benefits of Full Stack rust-webrtc
+
+Before now, the only way the browser could connect to a rust-libp2p server over WebRTC was to use [js-libp2p](https://github.com/libp2p/js-libp2p).
+
+However, full stack rust-libp2p WebRTC is so beneficial. First, troubleshooting was much more difficult to do in two stacks, two languages, two styles of implementation. Even though js-libp2p and rust-libp2p implement the same libp2p spec, it's much more difficult to switch between languages when troubleshooting bugs and data flow. With full-stack rust-libp2p, we can now build applications that run in the browser and on the server in rust, using the same language and style of implementation on both the client and the server. This makes both developing applications fatser and easier as we can re-use code in both the browser and on the server. In fact, we were able to move a lot of code out of the server crate into a [common WebRTC utilities crate within the rust-libp2p repo](https://github.com/libp2p/rust-libp2p/tree/master/misc/webrtc-utils), This will apply to libp2p applications as well!
+
+## How does libp2p use WebRTC?
+
+The bindings to [web-sys](https://docs.rs/web-sys/latest/web_sys/) are implemented as a transport, which means that you can connect to the libp2p network from the browser as you would any other transport in `rust-libp2p`. There is a full stack example [in the repo](https://github.com/libp2p/rust-libp2p/tree/master/examples/browser-webrtc).
```

Comment:
This paragraph is confusing to me. I think it is worth reintroducing web-sys here. How about something along the lines of:

```suggestion
## How does rust-libp2p in the browser use WebRTC?

- ~there is web-sys
- enables one to bind to browser apis
- rust-libp2p wraps web-sys, implementing the `Transport` trade
- Thus allows any application protocol (e.g. Kademlia) to make use of webrtc connections.

There is a full stack example [in the repo](https://github.com/libp2p/rust-libp2p/tree/master/examples/browser-webrtc).
```